plugins/colorschemes/kanagawa.nix

```diff
@@ -5,195 +5,232 @@
+                foo = "#RRGGBB",
+                bar = "#RRGGBB"
+              },
+              theme = {
+                foo = "#RRGGBB",
+                bar = "#RRGGBB"
               }
-            ```
-            It should return a table of highlights.
+            }
+          ```
+          It should return a table of highlights.
 
-            ```
-              function(colors)
-                CONTENT_OF_OVERRIDE_OPTION
-              end
-            ```
-          '';
+          ```
+            function(colors)
+              CONTENT_OF_OVERRIDE_OPTION
+            end
+          ```
+        '';
 
-        theme = helpers.defaultNullOpts.mkStr "wave" ''
-          The theme to load when background is not set.
+      theme = helpers.defaultNullOpts.mkStr "wave" ''
+        The theme to load when background is not set.
+      '';
+
+      background = {
+        light = helpers.defaultNullOpts.mkStr "lotus" ''
+          The theme to use when `vim.o.background = "light"`.
         '';
-        background = {
-          light = helpers.defaultNullOpts.mkStr "lotus" ''
-            The theme to use when `vim.o.background = "light"`.
-          '';
 
-          dark = helpers.defaultNullOpts.mkStr "wave" ''
-            The theme to use when `vim.o.background = "dark"`.
-          '';
-        };
+        dark = helpers.defaultNullOpts.mkStr "wave" ''
+          The theme to use when `vim.o.background = "dark"`.
+        '';
       };
-  };
-  config = let
-    setupOptions = with cfg;
-      {
-        inherit
-          compile
-          undercurl
-          commentStyle
-          functionStyle
-          keywordStyle
-          statementStyle
-          typeStyle
-          transparent
-          dimInactive
-          terminalColors
-          ;
-        colors = with colors; {
-          inherit theme palette;
+    };
+
+    settingsExample = {
+      compile = false;
+      undercurl = true;
+      commentStyle.italic = true;
+      functionStyle = {};
+      transparent = false;
+      dimInactive = false;
+      terminalColors = true;
+      colors = {
+        theme = {
+          wave.ui.float.bg = "none";
+          dragon.syn.parameter = "yellow";
+          all.ui.bg_gutter = "none";
         };
-        overrides =
-          helpers.ifNonNull' overrides
-          (helpers.mkRaw ''
-            function(colors)
-              ${overrides}
-            end
-          '');
-        inherit theme;
-        background = with background; {
-          inherit light dark;
+        palette = {
+          sumiInk0 = "#000000";
+          fujiWhite = "#FFFFFF";
         };
-      }
-      // cfg.extraOptions;
-  in
-    mkIf cfg.enable {
-      colorscheme = "kanagawa";
+      };
+      overrides = "function(colors) return {} end";
+      theme = "wave";
+    };
 
-      extraPlugins = [cfg.package];
+    extraConfig = cfg: {
+      warnings =
+        optional ((config.options.background or null) == null)
```

Comment:
I _think_ if colorschemes.kanagawa.settings.background is set to `emptyTable` _then_ background is not set. It is also possible to add 
```
vim.cmd("colorscheme kanagawa-wave")
vim.cmd("colorscheme kanagawa-dragon")
vim.cmd("colorscheme kanagawa-lotus")
```

manually to the extraConfigLua, but I guess this is outside of what we want to help.

The readme says:

Themes can be changed in three ways:

    Setting config.theme to the desired theme. Note that vim.o.background must be unset.
    Using the background option: Any change to the value of vim.o.background will select the theme mapped by config.background. Use vim.o.background = "" to unset this option.
    Loading the colorscheme directly with: 

This is what is telling me that the warning is not sufficient, as it only handles case 2.

Reply:
I planned to support cases 1 and 2, but not 3.
In both cases 1 and 2, background must be explicitly **set** (in the nixvim sense), to either `""`, `"dark"` or `"light"`.

Reply:
Hmmm.

Reply:
So, indeed, this module target use cases 1 and 2.
However the warning is useless. If the user doesn't set `options.background`, its default value is `"dark"` in neovim, and this is fine.